### ISSUE TITLE
SP4 Ticks: Layout does not resize to accommodate manual ticks

### ISSUE THREAD

**Issue by StefanBertels** [ScottPlot/ScottPlot#2603]
## Bug Report

**Issue:** 

Layout changed (text overlapping) in 4.1.63.
Everything worked fine in 4.1.62.

Annotation texts are printed all in the same line.
YAxis2 label overlaps with tick labels.

Probably related to #2302. 

Note: I'm ok with changes (if intended) but breaking changes like this should result in compiler errors (changed function signature etc.) or at least result in an increased minor version number (IMHO).

**Reproducing:** 

```cs
var plt = new ScottPlot.Plot(1600, 900);

plt.YAxis2.Dims.SetAxis(0, 25);
plt.YAxis2.Label("LABEL 0-5");
plt.YAxis2.Ticks(true);

plt.YAxis2.ManualTickPositions(new double[] {  11, 12, 13, 14 }, new[] { "eleven", "twelve", "thirteen", "fourteen" });

plt.AddAnnotation($"Text1 in line 1", 10, 10);
plt.AddAnnotation($"Text2 in line 2", 10, 30);
plt.AddAnnotation($"Text3 in line 2", 10, 50);

var stream = plt.GetImageBytes();
File.WriteAllBytes(@"c:\tmp\test.png", stream);
```

### System Details
* **ScottPlot Version:** 4.1.63
* **Operating System:** Windows 10
* **Application Type:** console
* **.NET Version:** .NET 7.0 (net7.0-windows)

## Test result (code above)
4.1.62 | 4.1.63
---|---
![scottplot_4 1 62](https://user-images.githubusercontent.com/3862727/233345724-89c0238c-af36-48be-91cf-9efc308f081d.png)|![scottplot_4 1 63](https://user-images.githubusercontent.com/3862727/233345747-5e82a3a1-d378-4add-9732-611f94cb2463.png)

**Comment by swharden** [comment]
Thanks for reporting this @StefanBertels (and @szescxz in #2605)

Definitely a bug (unrelated to annotation alignment though) and here's a more streamlined way to reproduce it:

```cs
ScottPlot.Plot plt = new(400, 300);

plt.YAxis2.Label("Secondary Vertical Axis");
plt.YAxis2.Ticks(true);

string lbl = "very long tick label";
plt.YAxis2.ManualTickPositions(
    positions: new double[] { -5, 0, 5 }, 
    labels: new string[] { lbl, lbl, lbl });

plt.SaveFig("issue2603.png");
```

![issue2603](https://github.com/ScottPlot/ScottPlot/assets/4165489/a496fd36-b732-4910-81e8-d0a011432fce)

I'll fix this before the next release 👍

**Comment by swharden** [comment]
Fixed in #2631

It was a bug introduced by a large refactor of the tick system last month https://github.com/ScottPlot/ScottPlot/commit/3f2aeadf969d5818d4d4f9cc70bcf4dfcb89f0ba
